dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/RequestItemRepository.java

```diff
@@ -0,0 +1,296 @@
+        throw new RepositoryMethodNotImplementedException(RequestItemRest.NAME, "delete");
+    }
+
+    @Override
+    @PreAuthorize("isAuthenticated()")
+    public RequestItemRest put(Context context, HttpServletRequest request,
+            String apiCategory, String model, String token, JsonNode requestBody)
+            throws AuthorizeException {
+        RequestItem ri = requestItemService.findByToken(context, token);
+        if (null == ri) {
+            throw new UnprocessableEntityException("Item request not found");
+        }
+
+        // Check for authorized user
+        RequestItemAuthor authorizer;
+        try {
+            authorizer = requestItemAuthorExtractor.getRequestItemAuthor(context, ri.getItem());
+        } catch (SQLException ex) {
+            LOG.warn("Failed to find an authorizer:  {}", ex.getMessage());
+            authorizer = new RequestItemAuthor("", "");
+        }
+        if (!authorizer.getEmail().equals(context.getCurrentUser().getEmail())) {
+            throw new AuthorizeException("Not authorized to approve this request");
+        }
+
+        // Make the changes
+        JsonNode acceptRequestNode = requestBody.findValue("acceptRequest");
+        if (null == acceptRequestNode) {
+            throw new UnprocessableEntityException("acceptRequest is required");
+        } else {
+            ri.setAccept_request(acceptRequestNode.asBoolean());
+        }
+
+        JsonNode responseMessageNode = requestBody.findValue("responseMessage");
+        String message = responseMessageNode.asText();
+
+        ri.setDecision_date(new Date());
+        requestItemService.update(context, ri);
+
+        // Send the response email
+        String subject = requestBody.findValue("subject").asText();
+        try {
+            RequestItemEmailNotifier.sendResponse(context, ri, subject, message);
+        } catch (IOException ex) {
+            throw new RuntimeException("Response not sent", ex);
+        }
+
+        RequestItemRest rir = requestItemConverter.convert(ri, Projection.DEFAULT);
+        return rir;
+    }
+
+    @Override
+    public Class<RequestItemRest> getDomainClass() {
+        return RequestItemRest.class;
+    }
+
+
+    /**
+     * Generate a link back to DSpace, to act on a request.
+     *
+     * @param token identifies the request.
+     * @return URL to the item request API, with the token as request parameter
+     *          "token".
+     * @throws URISyntaxException passed through.
+     * @throws MalformedURLException passed through.
+     */
+    private String getLinkTokenEmail(String token)
+            throws URISyntaxException, MalformedURLException {
+        String base = configurationService.getProperty("dspace.server.url");
+
+        URI link = new URIBuilder(base)
+                .setPath("/api/" + RequestItemRest.CATEGORY
+                        + '/' + RequestItemRest.PLURAL_NAME)
+                .addParameter("token", token)
+                .build();
```

Comment:
I think this link is incorrect, as we'll need to create a form/page on the UI for the user to read the request and respond.  This currently is a REST API link, which won't take the user anywhere useful.

I think you'll need to work with @artlowel to craft a UI URL where the user should be sent if they want to respond to an Item Request.  That UI URL will obviously need to be passed the token, so that it can lookup the request (via the REST API) and eventually craft the proper `PUT` to respond to the request.

Reply:
I do need to coordinate with the UI code, but it's a bit more complicated.  Since this link appears in an email, it will result in a GET request.  That means we need to link to the back-end, to bring up the UI at the approval page with the token pre-loaded.  THAT page will do the POST which accepts/rejects the request.  On the back-end, I need to see how similar things are done during user registration, and do likewise.  Thanks for reminding me that this part of the process is not yet coded.  Sorry that I requested re-review too soon!

Reply:
@mwoodiupui : I'm not sure I understand why the request-a-copy email needs to send a link to the backend, as that seems to imply you'd still need to redirect to the UI at some point.  

Instead, it seems like the flow here could be more similar to the Registration process, which sends you an email after you fill out a registration form.  In the email it sends, it links you back to the *User Interface* (see https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/eperson/AccountServiceImpl.java#L260), which then allows you to complete your registration. 

I *think* you might be able to do the same, if you agree on a URL path with @artlowel .  It could just be a path like `${dspace.ui.url}/request-a-copy?token={token}` ... and then we'd just need to add code to the UI to ensure when that path is called, it uses the `token` to load that request's information via a GET request to the backend.

Reply:
`${dspace.ui.url}/request-a-copy?token={token}` sounds just fine. Let's go with that.

Reply:
On second thought, let's make it `${dspace.ui.url}/request-a-copy/{token}` that way it'll work similar to the registration url, and it'll be slightly easier to parse in the UI

Reply:
So e.g. `https://demo7.dspace.org/request-a-copy/7177942B`